index.js

```diff
@@ -1,8 +1,12 @@
 const endable = require('./endable')
 
+// Indicates if object can be converted to a Buffer
+const isBufferCompatible = (o) => Buffer.isBuffer(o) || typeof o === 'string'
+
 module.exports = (stream, goodbye) => {
-  goodbye = Buffer.from(goodbye || 'GOODBYE')
+  goodbye = goodbye || 'GOODBYE'
```

Comment:
```suggestion
  goodbye = goodbye || Buffer.from('GOODBYE')
```

So, assume not "objectMode" by default.

Then, if goodbye is a buffer then we `Buffer.from(chunk)` for each chunk and do the slice check because the chunk may include [some other data + goodbye]. Otherwise we `===` compare the chunk with `goodbye` ("objectMode").

Basically, you put it into "objectMode" by passing a non-buffer `goodbye`.

What do you think?

Reply:
I wonder how often people will want to use a goodbye that's neither a Buffer nor a string?

Reply:
No idea, but it should be possible right?

I'm trying to avoid doing the detection for every chunk and if the chunks are strings, doing the conversion to buffer and slicing for every chunk.

Reply:
Makes sense 👍 
I'll take a look on Monday